warehouse/migrations/versions/a78d021addb1_record_if_macaroon_predates_permission_.py

```diff
@@ -0,0 +1,50 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Record if a Macaroon predates the permission caveat
+
+Revision ID: a78d021addb1
+Revises: 812e14a4cddf
+Create Date: 2024-01-20 03:29:32.962023
+"""
+
+import sqlalchemy as sa
+
+from alembic import op
+
+revision = "a78d021addb1"
+down_revision = "812e14a4cddf"
+
+
+def upgrade():
+    # Initially set our default to true, so all existing macaroons get this set
+    # to true, then change it so new macaroons get it set to false.
+    op.add_column(
```

Comment:
Not sure about the locking side, but thinking out loud for the row count: we can probably purge 99% of those OIDC generated tokens, since they should all be expired and (IIRC) shouldn't need to be persisted for any UI state (since the relevant bits are separately persisted via the `Event` model payloads).

Reply:
@woodruffw irrespective of this PR, it'd be great to have a daily task run to expire/purge useless OIDC macaroons from the DB to prevent growth

Reply:
100%! I'll file a separate tracking issue for that.